### DISCUSSION TITLE
Help Needed: Accelerating Custom U-Net CNN on PYNQ Z2 Using FINN

### DISCUSSION THREAD

**Discussion by taharabs** [Xilinx/finn#1096] [Q&A]
Hi everyone,

I'm working on accelerating a custom CNN model using the PYNQ Z2 board. I initially looked into using Vitis AI, but I found out that it doesn't support the Zynq 7000 series. I was advised to use FINN to compile my CNN as a dataflow accelerator.

However, I’m encountering some confusion about the process. Specifically, I’m unclear about:

1. Why can't I compile and run the dataflow accelerator directly on the PYNQ Z2 board?
2. What steps should I follow to successfully compile and deploy my custom CNN model on the PYNQ Z2 using FINN?
3. there any specific resources or documentation that can guide me through this process?
Any guidance, resources, or step-by-step instructions to help me understand and solve this issue would be greatly appreciated.

Thank you for your help!

tahar

**Comment by fpjentzsch** [comment]
Hi,

1) Accelerator compilation involves high-level synthesis with Vitis HLS and synthesis with Vivado. Thus, you need to run FINN on a x86 Ubuntu machine.
2) Train a quantized model using Brevitas, export it as QONNX, run a FINN dataflow build on it, deploy it on the board with the generated Python driver.
2) Install FINN via the Getting Started guide in the documentation, then reproduce the build of a similar example model from the finn-examples repo and go from there.
